### PR TITLE
Add first and last helper methods for record retrieval

### PR DESCRIPTION
**Pull request by cdesch**
First and last helper methods will allow for records frequently looked for at the top or bottom of a list to be quickly retrieved. Included in the pull request are test cases for the helper methods that can be run using automatically via the roboeletric/travis-ci automated testing. 

E.g. 

     MyObject myFirstObjectInstance = SugarRecord.first(MyObject.Class);
     MyObject myLastObjectInstance = SugarRecord.last(MyObject.Class);

instead of:

     MyObject myFirstObjectInstance = SugarRecord.findById(MyObject.Class, 1);
     MyObject myLastObjectInstance = SugarRecord.findById(MyObject.class, SugarRecord.count(MyObject.class));

Feedback is always welcome.

### CODE REVIEW

File: library/src/main/java/com/orm/SugarRecord.java [hunk 1]
```diff
@@ -75,6 +75,14 @@
         return findById(type, Long.valueOf(id));
     }
 
+    public static <T> T first(Class<T>type){
+        return findById(type, 1);
```
Comment (whoshuu) [inline library/src/main/java/com/orm/SugarRecord.java]: The first object in the database may not necessarily have an `id` of `1`. I think a better implementation would do a find with an order by ascending `id` and limit the query to a single item.

File: library/src/main/java/com/orm/SugarRecord.java [hunk 2]
```diff
@@ -75,6 +75,14 @@
...
+    }
+
+    public static <T> T last(Class<T>type){
+        return findById(type, count(type));
```
Comment (whoshuu) [inline library/src/main/java/com/orm/SugarRecord.java]: As above, the item with an `id` equal to the total number of records might not be the last record if some objects were deleted from the database.

It'd be nice to actually capture both of these failing cases in your tests, and solve them in your implementations of `first` and `last`. If you need help writing those test cases, let me know and I'll throw in a commit!

**Comment by cdesch** [comment]
thats right! I was so focused on the test cases I forgot that the record `id`'s aren't the same as in arrays. I'll rewrite the helper methods and the test cases. Thanks.

**Comment by cdesch** [comment]
I'm changing the methods to use `findWithQuery` although I ran into the issue #294 .


Here is my updated code thus far. Feedback welcome!!

    public static <T> T first(Class<T>type){
        List<T> list = findWithQuery(type, "SELECT * FROM ? ORDER BY ROWID ASC LIMIT 1", NamingHelper.toSQLName(type));
        if (list.isEmpty()) return null;
        return list.get(0);
    }

    public static <T> T last(Class<T>type){
        List<T> list = findWithQuery(type, "SELECT * FROM ? ORDER BY ROWID DESC LIMIT 1", NamingHelper.toSQLName(type));
        if (list.isEmpty()) return null;
        return list.get(0);
    }

**Comment by cdesch** [comment]
@whoshuu I've updated the pull request with changes requested and test cases.

**Comment by whoshuu** [comment]
I'll take a look at this tonight and try to get it in then.

**Pull request closed without merging**